src/actions/entity.js

```diff
@@ -140,76 +140,263 @@ export function cloneSelectedEntity () {
  * @return {string}        Entity clipboard representation
  */
 export function getClipboardRepresentation (entity) {
-  // entity.flushToDOM();
-  var clone = entity.cloneNode(true);
-  var defaultComponents = Object.keys(clone.defaultComponents);
-
-  removeDefaultAttributes(clone);
-  removeNotModifiedMixedinAttributes(entity, clone);
-  removeDefaultProperties(entity, clone);
-  clone.flushToDOM();
+  var clone = prepareForSerialization(entity);
+  clone.flushToDOM(true);
   return clone.outerHTML;
+}
 
-  function removeDefaultAttributes (el) {
-    for (let i = 0; i < el.childNodes.length; i++) {
-      var child = el.childNodes[i];
-      if (child.isEntity) {
-        removeDefaultAttributes(child);
-      }
+/**
+ * Returns a copy of the DOM hierarchy prepared for serialization.
+ * The process optimises component representation to avoid values coming from
+ * primitive attributes, mixins and defaults.
+ *
+ * @param {Element} entity Root of the DOM hierarchy.
+ * @return {Elment}        Copy of the DOM hierarchy ready for serialization.
+ */
+function prepareForSerialization(entity) {
+  var clone = entity.cloneNode(false);
+  var children = entity.childNodes;
+  for (var i = 0, l = children.length; i < l; i++) {
+    var child = children[i];
+    if (child.nodeType !== Node.ELEMENT_NODE ||
+        !child.hasAttribute('aframe-injected') &&
```

Comment:
We should include `data-aframe-inspector` too, as it's used for the elements added by the inspector